bindings/experimental/distrdf/python/DistRDF/Backends/Dask/Backend.py

```diff
@@ -63,8 +63,14 @@ def ProcessAndMerge(self, ranges, mapper, reducer):
             after computation (Map-Reduce).
         """
 
-        # With the Spark backend these need to be passed as variables.
-        # Doing the same here for coherency, might change in the future.
+        # These need to be passed as variables, because passing `self` inside
+        # following `dask_mapper` function would trigger serialization errors
+        # like the following:
+        #
+        # AttributeError: Can't pickle local object 'DaskBackend.ProcessAndMerge.<locals>.dask_mapper'
```

Comment:
So indeed this is also a problem with Dask

Reply:
Yes indeed also the Dask client holds some unserializable state so it can't be pickled together with the modified mapper function